src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/SampleNameMap.java

```diff
@@ -0,0 +1,274 @@
+     */
+    public void addSample(final String sampleName, final URI vcfPath) {
+        addSample(sampleName, vcfPath, null);
+    }
+
+    /**
+     * Add a new sample mapping
+     *
+     * @param sampleName name of the sample
+     * @param vcfPath path to the VCF for the sample (not null)
+     * @param vcfIndexPath path to the index for the sample (may be null)
+     */
+    public void addSample(final String sampleName, final URI vcfPath, final URI vcfIndexPath) {
+        if ( ! sampleNameIsLegal(sampleName) ) {
+            throw new UserException.BadInput("Sample name " + sampleName + " is not legal. Sample names must be non-empty and not contain leading or trailing whitespace");
+        }
+        if ( vcfPath == null ) {
+            throw new UserException.BadInput("VCF path for sample " + sampleName + " was null");
+        }
+
+        final URI previousPath = sampleNameToVcfPath.put(sampleName, vcfPath);
+        if (previousPath != null) {
+            throw new UserException.BadInput("Duplicate sample: " + sampleName + ". Sample was found in both "
+                    + vcfPath + " and " + previousPath + ".");
+        }
+
+        if (vcfIndexPath != null) {
+            final URI previousIndexPath = sampleNameToVcfIndexPath.put(sampleName, vcfIndexPath);
+            if (previousIndexPath != null) {
+                throw new UserException.BadInput("For sample " + sampleName + ", attempted to specify multiple indices: " + vcfIndexPath + " and " + previousIndexPath);
+            }
+        }
+    }
+
+    /**
+     * @return The full mapping of sample names -> VCF paths, with the sample names in sorted order
+     */
+    public SortedMap<String, URI> getSampleNameToVcfPath() {
+        return sampleNameToVcfPath;
+    }
+
+    /**
+     * @param sample sample name
+     * @return the VCF associated with that sample name, as a URI
+     */
+    public URI getVCFForSample(final String sample) {
+        return sampleNameToVcfPath.get(sample);
+    }
+
+    /**
+     * @param sample sample name
+     * @return the VCF associated with that sample name, as a Path
+     */
+    public Path getVCFForSampleAsPath(final String sample) {
+        final URI vcfURI = sampleNameToVcfPath.get(sample);
+        return vcfURI == null ? null : IOUtils.getPath(vcfURI.toString());
+    }
+
+    /**
+     * @param sample sample name
+     * @return the VCF index associated with that sample name, as a URI, or null if no index
+     */
+    public URI getVCFIndexForSample(final String sample) {
+        return sampleNameToVcfIndexPath.get(sample);
+    }
+
+    /**
+     * @param sample sample name
+     * @return the VCF index associated with that sample name, as a Path, or null if no index
+     */
+    public Path getVCFIndexForSampleAsPath(final String sample) {
+        final URI vcfIndexURI = sampleNameToVcfIndexPath.get(sample);
+        return vcfIndexURI == null ? null : IOUtils.getPath(vcfIndexURI.toString());
```

Comment:
This is so much gross.  URI -> String -> Path back through UR with a side of null checking.  Nothing to be done :(